File: JobQuestRestBackend/src/main/java/com/cybage/boot/models/UserWorkExperience.java
package com.cybage.boot.models;

import jakarta.persistence.*;
import lombok.*;

@Data
@Entity
@Table(name = "userWorkExperience")
@AllArgsConstructor
@NoArgsConstructor
public class UserWorkExperience {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private int workExpId;
	private String companyName;
	private String titleName;
	private String startDate;
	private String endDate;
	private String workDescription;
	@ManyToOne
	@JoinColumn(name = "user_id")
	private UserProfileModel user_id; // Name must Match with MappedBy Value
	// private int user_id; // FOREIGN KEY
}

File: JobQuestFrontend/src/app/models/userWorkExperienceModel/user-work-experience.ts
import { UserProfile } from '../userProfile/user-profile';

export class UserWorkExperience {
  workExpId: number;
  companyName: string;
  titleName: string;
  startDate: string;
  endDate: string;
  workDescription: string;
  user_id: UserProfile;

  constructor(
    workExpId: number,
    companyName: string,
    titleName: string,
    startDate: string,
    endDate: string,
    workDescription: string,
    user?: UserProfile
  ) {
    this.workExpId = workExpId;
    this.companyName = companyName;
    this.titleName = titleName;
    this.startDate = startDate;
    this.endDate = endDate;
    this.workDescription = workDescription;
    this.user_id = user || new UserProfile();
  }
}
